frontend/src/util/stores/openingStore.tsx

```diff
@@ -0,0 +1,27 @@
+import { create } from 'zustand';
+
+interface OpeningDetails {
+    id: number | null;
+    recruitment_round_id: number | null;
+    student_team_name: string | null;
+    title: string | null;
+    application_count: number | null;
+}
+
+interface OpeningStore {
+    selectedOpening: OpeningDetails | null;
+    setSelectedOpening: (opening: OpeningDetails) => void;
+    clearSelectedOpening: () => void;
+}
+
+export const useOpeningStore = create<OpeningStore>((set) => ({
+    selectedOpening: {
+        id: null,
+        recruitment_round_id:  null,
+        student_team_name: null,
+        title: null,
+        application_count: null,
+    },
+    setSelectedOpening: (opening) => set({ selectedOpening: opening }),
+    clearSelectedOpening: () => set({ selectedOpening: null }),
```

Comment:
does it matter if we always update the store when entering a new page

Reply:
Well it would be consistent with the rest of the application and easy to debug if the store is always cleared when backing up. But if you decide it's not that important, just delete the clearSelectedOpening method ig